inline_scan.sh

```diff
@@ -403,51 +403,44 @@ get_repo_digest_id() {
     REPO=$(echo ${IMAGE_NAMES[0]} | rev |  cut -d / -f 2 | rev)
     BASE_IMAGE=$(echo ${IMAGE_NAMES[0]} | rev | cut -d / -f 1 | rev | cut -d : -f 1)
     TAG=$(echo ${IMAGE_NAMES[0]} | rev | cut -d / -f 1 | rev | cut -d : -f 2)
+    REPO_TAG=$(docker inspect --format="{{ index .RepoTags 0 }}" "${SCAN_IMAGES[0]}")
 
     if [[ -z "${TAG// }" ]]; then
         TAG='latest'
     fi
 
-    FINAL_DIGEST="sha256@12345"
-    for DIGEST in "${DIGESTS[@]}"
-    do
-        if [[ ${DIGEST} == *"${REPO}/${BASE_IMAGE}:${TAG}"* || ${DIGEST} == *"${REPO}/${BASE_IMAGE}"* || ${DIGEST} == *"${BASE_IMAGE}"* ]]; then
-            FINAL_DIGEST=$(echo ${DIGEST} | rev | cut -d : -f 1 | rev | tr -d ']' | cut -d ' ' -f 1)
-        else
-            printf '%s\n' " Unable to compute the digest from docker inspect ${SCAN_IMAGES[0]}!"
-            printf '%s\n' " Consider running with -d option with a valid sha256:<digestID>."
-        fi
-    done
+    REPO_DIGEST=$(docker inspect --format="{{- if .RepoDigests -}}{{index .RepoDigests 0}}{{- else -}}{{- end -}}" ${SCAN_IMAGES[0]} | cut -d "@" -f 2 | cut -c 8-)
 
     # Generate Image digest ID for given image, if repo digest is not present
-    if [[ "${SYSDIG_IMAGE_DIGEST}" == 'sha256:123456890abcdefg' ]]; then
+    if [[ -z "${REPO_DIGEST}" ]]; then
```

Comment:
I've changed the logic here: if I can't get the digest via docker inspect, then I generate one.

Reply:
@ProvoK This makes sense from a cleaner implementation perspective, but the reason we nudged the customer to pass a digest themselves, so that they can control how it's searched by them later in the UI/API.
@mattpag @yathi-sysdig @marojor What we need to discuss is that, even this change as such can have a scenario where User -> scans an image with no digest (before uploading to registry) . Then once it passes, scans at a later time the same image (with no content changes) but will have a different digestId preset now. This was the case always, but want to make sure, that's ok now as well.